src/plugins/saved_objects/public/saved_object/helpers/hydrate_index_pattern.ts

```diff
@@ -31,25 +31,19 @@ export async function hydrateIndexPattern(
   indexPatterns: IndexPatternsContract,
   config: SavedObjectConfig
 ) {
-  const clearSavedIndexPattern = !!config.clearSavedIndexPattern;
```

Comment:
I couldn't find any reference to this functionality, seems like this is safe to remove